build/lib/src/analyzer/resolver.dart

```diff
@@ -16,7 +16,8 @@ abstract class Resolver {
   /// or is a `part of` file (not a standalone Dart library).
   Future<bool> isLibrary(AssetId assetId);
 
-  /// All libraries accessible from the entry point, recursively.
+  /// All libraries recursively accessible from the entry point or subsequent
+  /// calls to [libraryFor] and [isLibrary].
```

Comment:
We should also add a similar note to `findLibraryByName` that it can only find libraries that have been previously resolved (or are transitive imports of one)